File: ComputerBasics/Programs/java/ClassDog.java
// Define a class named 'Dog'
class Dog {
    // Attributes (fields) of the Dog class
    String name;
    String breed;

    // Method to make the dog bark
    void bark() {
        System.out.println(name + " says Woof! Woof!");
    }

    // Method to display the dog's details
    void displayDetails() {
        System.out.println("Name: " + name + ", Breed: " + breed);
    }
}

// Main class to demonstrate the Dog class
public class ClassDog {
    public static void main(String[] args) {
        // Create an object (instance) of the Dog class
        Dog myDog = new Dog();

        // Assign values to the object's attributes
        myDog.name = "Buddy";
        myDog.breed = "Golden Retriever";

        // Call methods on the object
        myDog.displayDetails(); // Output: Name: Buddy, Breed: Golden Retriever
        myDog.bark();           // Output: Buddy says Woof! Woof!

        // Create another object of the Dog class
        Dog anotherDog = new Dog();
        anotherDog.name = "Max";
        anotherDog.breed = "Labrador";

        anotherDog.displayDetails(); // Output: Name: Max, Breed: Labrador
        anotherDog.bark();           // Output: Max says Woof! Woof!
    }
}

File: ComputerBasics/Programs/Python/ClassDog.py
class Dog:
    # The __init__ method is a constructor, automatically called when a new object is created.
    # 'self' refers to the instance of the class (the object itself).
    def __init__(self, name, breed):
        self.name = name  # Instance attribute
        self.breed = breed  # Instance attribute

    # This is a method (a function defined within a class).
    # It takes 'self' as its first parameter to access the object's attributes.
    def bark(self):
        return f"{self.name} says Woof!"

    def describe(self):
        return f"{self.name} is a {self.breed}."

# Create objects (instances) of the Dog class
my_dog = Dog("Buddy", "Golden Retriever")
another_dog = Dog("Lucy", "Beagle")

# Access attributes of the objects
print(f"My dog's name is {my_dog.name} and it's a {my_dog.breed}.")
print(f"Another dog's name is {another_dog.name} and it's a {another_dog.breed}.")

# Call methods on the objects
print(my_dog.bark())
print(another_dog.describe())